src/thunderbots/software/ai/hl/stp/action/chip_action.cpp

```diff
@@ -0,0 +1,110 @@
+#include "ai/hl/stp/action/chip_action.h"
+
+#include "ai/intent/chip_intent.h"
+#include "ai/intent/move_intent.h"
+#include "geom/polygon.h"
+#include "geom/util.h"
+
+ChipAction::ChipAction(double length_of_region_behind_ball)
+    : Action(), length_of_region_behind_ball(length_of_region_behind_ball)
+{
+}
+
+std::unique_ptr<Intent> ChipAction::updateStateAndGetNextIntent(
+    const Robot& robot, Point chip_origin, Point chip_target, double chip_distance_meters)
+{
+    return updateStateAndGetNextIntent(robot, chip_origin,
+                                       (chip_target - chip_origin).orientation(),
+                                       chip_distance_meters);
+}
+
+std::unique_ptr<Intent> ChipAction::updateStateAndGetNextIntent(
+    const Robot& robot, Point chip_origin, Angle chip_direction,
+    double chip_distance_meters)
+{
+    // Update the parameters stored by this Action
+    this->robot                = robot;
+    this->chip_origin          = chip_origin;
+    this->chip_direction       = chip_direction;
+    this->chip_distance_meters = chip_distance_meters;
+
+    return getNextIntent();
+}
+
+std::unique_ptr<Intent> ChipAction::calculateNextIntent(
+    intent_coroutine::push_type& yield)
+{
+    // TODO: replace with real constant
+    double DIST_TO_FRONT_OF_ROBOT = 0.05;
+    // ASCII art showing the region behind the ball
+    //
+    //              X
+    //       v-------------v
+    //
+    //    >  \-------------/
+    //    |   \           /
+    //    |    \         /
+    //    |     \       /       <- Region considered "behind ball"
+    // 2X |      \     /
+    //    |       \   /
+    //    |        \ /
+    //    >         v
+    //
+    //
```

Comment:
what's this distance (triangle to ball)?

Reply:
Added distance to diagram